src/documents/components/DataValidation/DataValidation.js

```diff
@@ -0,0 +1,244 @@
+import {
```

Comment:
@XuTheBunny Sorry for the long comment wasn't sure where to put all this. Also, I'm not sure if you were planning to implement some of this later once more of the validation was implemented on the backend. If so, I apologize for leaving feedback you might not need lol. But just in case:

**Validation Running**
User clicks the Click to Start Validation button, and then clicks the Run Validation button on the Validation view. Then the Running validation loader pops up. When the user navigates away from the Validation view while the validation is still running, the Data Review detail page still shows Click to Start Validation. Perhaps it should show a Running badge? Also, when the user clicks the Click to Start Validation button while the previous validation is running, the Validation view still shows the Run Validation button.

**Validation Failures**
In general it seems like the validation run failed state isn't taken into account yet. For example, the validation view doesn't update upon a validation run failure. It stays on the running validation loader. I think it would be useful to display the error message upon failure and then have a Re-run validation button somewhere.

**Data Review Detail View**
The data review detail view only shows the validation run states when the run is not started (Click To Start Validation) or when the run successfully completed (shows the result summary). Wondering if that area on the UI should also show the other states: failed, canceleld?

**Cancel Validation**
There doesn't seem to be a way to cancel a validation. Will this be added later?

**Re-run Validation**
There doesn't seem to be a way to re-run a failed validation. Will this be added later?